### PR TITLE
[no-release-notes] improve explain output for `IndexTableAccess` with dynamic lookups

### PR DESCRIPTION
**Pull request by JCOR11599**
This PR changes IndexedTableAccess to print the keys used during dynamic lookups for better plan readability.
Additionally, this PR contains further testing for an optimization over LookupJoins, specifically with pointlookups.

### CODE REVIEW

File: sql/plan/indexed_table_access.go [hunk 1]
```diff
@@ -256,6 +256,14 @@ func (i *IndexedTableAccess) String() string {
 		}
 	}
 
+	if i.lb != nil && len(i.lb.keyExprs) > 0 {
```
Comment (max-hoffman) [inline sql/plan/indexed_table_access.go:259]: we should also add to DebugString()

File: enginetest/queries/query_plans.go [hunk 2]
```diff
@@ -10629,3 +10637,128 @@ order by i;`,
...
+  t2
+on
+  t1.id = t2.id and t1.a = t2.a
+inner join
+  t3
+on
+  t3.id = t2.id and t3.b = t2.b
+where
+  t2.id = "id1" and t2.b = "b1";`,
+				ExpectedPlan: "Project\n" +
```
Comment (max-hoffman) [inline enginetest/queries/query_plans.go:10707]: we should make these autogen-able somehow, or in a format that can auto regen

**Review by max-hoffman — COMMENTED** [review]
only blocking comment is having plan in a format that can autogen

File: sql/plan/indexed_table_access.go (hunk 1 above)
Comment (JCOR11599) [reply sql/plan/indexed_table_access.go:259]: It was already in `DebugString()`

**Review by max-hoffman — APPROVED** [review]
lgtm, thanks for doing a second pass to simplify the test

**Pull request merged**